src/nanorc/common_commands.py

```diff
@@ -328,30 +328,49 @@ def add_common_cmds(shell, end_of_run_cmds=True):
     # shell.add_command(ls_thread       , 'ls_thread'       )
 
 def add_custom_cmds(cli, rc_cmd_exec, cmds):
+    '''
     for c,d in cmds.items():
         arg_list = {}
         arg_default = {}
         for app_data in d:
-            for modules_data in app_data.values():
-                for module_data in modules_data:
-                    module = module_data['match']
-                    cmd_data = module_data.get('data')
-                    if not cmd_data: continue
-                    for arg in cmd_data:
-                        arg_list[arg] = type(cmd_data[arg])
-                        arg_default[arg] = cmd_data[arg]
+            if type(app_data) == dict:
+                for modules_data in app_data.values():
+                    if type(modules_data) == dict:
+                        for module_data in modules_data:
+                            if 'match'  not in module_data:
+                                continue
+                            module = module_data['match']
+                            cmd_data = module_data.get('data')
+                            if not cmd_data: continue
+                            for arg in cmd_data:
+                                arg_list[arg] = type(cmd_data[arg])
+                                arg_default[arg] = cmd_data[arg]
+    '''
+    for cmd_name, cmd_data in cmds.items():
+        arg_list = {}
+        arg_default = {}
+        if type(cmd_data) != dict: continue
+        for app_name, app_data in cmd_data.items():
+            for module in app_data['modules']:
+                arg_data = module.get('data')
+                if not arg_data: continue
+                for arg in arg_data:
```

Comment:
Isn't this data a dictionary? Shouldn't it be iterated over with a `items()` or at least `keys()`?

Reply:
app_data['modules'] is a list, since the modules don't seem to have names.